integrated_channel_plugin_sapsf/models.py

```diff
@@ -0,0 +1,135 @@
+
+
+# Create your models here.
+
+@python_2_unicode_compatible
+class OCNWebServicesConfiguration(TimeStampedModel):
+    """
+    The Global configuration for integrating with SuccessFactors.
+    """
+
+    completion_status_api_path = models.CharField(max_length=255)
+    course_api_path = models.CharField(max_length=255)
+    oauth_api_path = models.CharField(max_length=255)
+    provider_id = models.CharField(max_length=100, default='EDX')
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        return "<OCNWebServicesConfiguration with id {id}>".format(
+            id=self.id
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class OCNWebServicesEnterpriseCustomerConfiguration(TimeStampedModel):
+    """
+    The Enterprise specific configuration we need for integrating with SuccessFactors.
+    """
+
+    enterprise_customer_uuid = models.UUIDField(unique=True)
+    sapsf_base_url = models.CharField(max_length=255)
+    key = models.TextField(blank=True, verbose_name="Client ID")
+    secret = models.TextField(blank=True, verbose_name="Client Secret")
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        return "<OCNWebServicesEnterpriseCustomerConfiguration for Enterprise {enterprise_uuid}>".format(
+            enterprise_uuid=self.enterprise_customer_uuid
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class CompletionStatusEventAudit(models.Model):
+    """
+    The payload we sent to SuccessFactors at a given point in time for an enterprise course enrollment.
+    """
+
+    enterprise_course_enrollment_id = models.PositiveIntegerField(blank=False, null=False)
+    sapsf_user_id = models.CharField(max_length=255, blank=False, null=False)
+    course_id = models.CharField(max_length=255, blank=False, null=False)
+    course_completed = models.BooleanField(default=True)
+    completed_timestamp = models.BigIntegerField()  # we send a UNIX timestamp to SuccessFactors
+    instructor_name = models.CharField(max_length=255, blank=True)
+    grade = models.CharField(max_length=100, blank=False, null=False)
+    error_message = models.TextField(blank=True)
+    created = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        return '<CompletionStatusEventAudit {} for enterprise enrollment {}, SAP user {}, and course {}>'.format(
```

Comment:
I was copying what I have seen in the enterprise models, so it might be a broader discussion to introduce a mixin, I don't have a strong opinion about this either way.